crates/interpreter/src/side_table.rs

```diff
@@ -14,26 +14,39 @@
 
 use alloc::vec::Vec;
 use core::ops::Range;
+use std::marker::PhantomData;
 
 use crate::error::*;
 use crate::module::Parser;
+use crate::toctou::*;
 
 pub const SECTION_NAME: &str = "wasefire-sidetable";
 
 #[derive(Debug)]
-pub struct SideTableView<'m> {
+pub struct SideTableView<'m, M: Mode> {
```

Comment:
Only the functions should be generic. The `SideTableView<'m>` doesn't need to be. It's a bit different from `Parser`.